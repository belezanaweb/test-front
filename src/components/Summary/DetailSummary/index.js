import React from 'react'
import PropTypes from 'prop-types'

import { Container, Title, Value } from './style'

const DetailSummary = ({ title, value, isHighlight, isBold }) => (
  <Container isHighlight={isHighlight} isBold={isBold}>
    <Title>{title}</Title>
    <Value>{value}</Value>
  </Container>
)

DetailSummary.defaultProps = {
  isHighlight: false,
  isBold: false
}

DetailSummary.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  isHighlight: PropTypes.bool,
  isBold: PropTypes.bool
}

export default DetailSummary