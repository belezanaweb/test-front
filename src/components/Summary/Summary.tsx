import { PropsWithChildren, ReactNode } from 'react'
import useCart from '../../hooks/useCart'
import Spinner from '../Spinner'
import MonetaryValue from '../MonetaryValue'

function SummaryRow({ children }: PropsWithChildren) {
  return <div className="flex justify-between mb-2">{children}</div>
}

export default function Summary({ renderAction }: { renderAction: () => ReactNode }) {
  const { data, isLoading, isError } = useCart()

  return (
    <section className="w-full shadow-md bg-white px-5 py-7 fixed bottom-0">
      {isLoading && <Spinner />}
      {!isLoading && (isError || !data) && (
        <div className="text-sm">Não foi possível carregar as informações do carrinho.</div>
      )}
      {!isLoading && !isError && data && (
        <div className="text-sm">
          <SummaryRow>
            <span>Produtos: ({data.items.length} itens)</span>
            <MonetaryValue value={data.subTotal} />
          </SummaryRow>
          <SummaryRow>
            <span>Frete:</span>
            <MonetaryValue value={data.shippingTotal} />
          </SummaryRow>
          <SummaryRow>
            <span>Desconto:</span>
            <MonetaryValue className="text-fuschia font-bold" value={data.discount} />
          </SummaryRow>
          <SummaryRow>
            <span className="font-bold text-base">Subtotal</span>
            <MonetaryValue className="font-bold text-base" value={data.total} />
          </SummaryRow>
        </div>
      )}
      <div className="mt-7">{renderAction()}</div>
    </section>
  )
}
