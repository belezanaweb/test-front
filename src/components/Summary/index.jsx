import React from 'react'

export const Summary = () => {
  return <div>Summary</div>
}
