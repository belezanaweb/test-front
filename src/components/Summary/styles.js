import styled from 'styled-components'

export const Container = styled.section`
  width: 100%;
  border: 1px solid #d4d4d4;
  padding: 15px;

  div {
    width: 100%;
    font-size: 0.875rem;
    color: #212122;
    display: flex;
    justify-content: space-between;

    &:not(:last-child) {
      margin-bottom: 8px;
    }

    &:nth-child(3) {
      margin-bottom: 16px;
    }
  }
`

export const Discount = styled.div`
  p {
    color: #ff7800;
  }
`
export const Total = styled.div`
  p {
    font-weight: 800;
  }
`
