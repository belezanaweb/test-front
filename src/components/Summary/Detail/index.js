import React from 'react'
import PropTypes from 'prop-types'

import { Container, Title, Value } from './styles'

const Detail = ({ title, value, isHighlight, isBold }) => (
  <Container isHighlight={isHighlight} isBold={isBold}>
    <Title>{title}</Title>
    <Value>{value}</Value>
  </Container>
)

Detail.defaultProps = {
  isHighlight: false,
  isBold: false
}

Detail.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  isHighlight: PropTypes.bool,
  isBold: PropTypes.bool
}

export default Detail
