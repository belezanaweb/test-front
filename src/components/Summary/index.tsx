import React from 'react';
import { Container, SummaryItem } from './styles';

interface SummaryProps {
  subTotal: string;
  shippingTotal?: string;
  discount?: string;
  total: string;
}

const Summary: React.FC<SummaryProps> = ({
  subTotal,
  shippingTotal,
  discount,
  total,
}) => {
  return (
    <Container>
      <SummaryItem>
        Produtos
        <span>{subTotal}</span>
      </SummaryItem>
      {shippingTotal && (
        <SummaryItem>
          Frete
          <span>{shippingTotal}</span>
        </SummaryItem>
      )}
      {discount && (
        <SummaryItem highlight>
          Desconto
          <span>{discount}</span>
        </SummaryItem>
      )}
      <SummaryItem bolder>
        Total
        <span>{total}</span>
      </SummaryItem>
    </Container>
  );
};

export default Summary;
