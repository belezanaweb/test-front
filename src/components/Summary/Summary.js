import React from 'react'
import { SummaryContainer, SummaryItem, SummaryText, SummaryTextBold } from './Summary.elements'

export default function Summary() {
  return (
    <section>
      <SummaryContainer>
        <SummaryItem>
          <SummaryText>PRODUTOS</SummaryText>
          <SummaryText>R$ 624,80</SummaryText>
        </SummaryItem>
        <SummaryItem>
          <SummaryText>FRETE</SummaryText>
          <SummaryText>R$ 624,80</SummaryText>
        </SummaryItem>
        <SummaryItem>
          <SummaryText discount>DESCONTO</SummaryText>
          <SummaryText discount>R$ 624,80</SummaryText>
        </SummaryItem>
        <SummaryItem>
          <SummaryTextBold>TOTAL</SummaryTextBold>
          <SummaryTextBold>R$ 624,80</SummaryTextBold>
        </SummaryItem>
      </SummaryContainer>
    </section>
  )
}
