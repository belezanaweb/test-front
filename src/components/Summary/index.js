import React from 'react'

import Detail from './Detail'

import { Container } from './styles'

const Summary = () => (
  <Container>
    <Detail title='Produtos' value='R$ 624,80' />
    <Detail title='Frete' value='R$ 5,30' />
    <Detail title='Desconto' value='- R$ 30,00' isHighlight />
    <Detail title='Total' value='R$ 600,10' isBold />
  </Container>
)

export default Summary
