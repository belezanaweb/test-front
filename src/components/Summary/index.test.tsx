import { render } from '@testing-library/react';
import React from 'react';
import Summary from '.';

const prices = {
  subTotalFormatted: 'R$ 610,00',
  shippingTotalFormatted: 'R$ 0,00',
  discountFormatted: 'R$ 10,00',
  totalFormatted: 'R$ 600,00',
};

describe('<Summary />', () => {
  it('should be able to render correctly', () => {
    // Arrange
    const { baseElement } = render(
      <Summary
        subTotal={prices.subTotalFormatted}
        shippingTotal={prices.shippingTotalFormatted}
        discount={prices.discountFormatted}
        total={prices.totalFormatted}
      />,
    );

    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  it('should be able to render correctly when not shipping', () => {
    // Arrange
    const { baseElement } = render(
      <Summary
        subTotal={prices.subTotalFormatted}
        discount={prices.discountFormatted}
        total={prices.totalFormatted}
      />,
    );

    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  it('should be able to render correctly when not discount', () => {
    // Arrange
    const { baseElement } = render(
      <Summary
        subTotal={prices.subTotalFormatted}
        shippingTotal={prices.shippingTotalFormatted}
        total={prices.totalFormatted}
      />,
    );

    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  it('should be able to render correctly when not shipping and discount', () => {
    // Arrange
    const { baseElement } = render(
      <Summary
        subTotal={prices.subTotalFormatted}
        total={prices.totalFormatted}
      />,
    );

    // Assert
    expect(baseElement).toMatchSnapshot();
  });
});
