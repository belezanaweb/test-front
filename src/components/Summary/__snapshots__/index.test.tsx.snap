// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> should be able to render correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bdnylx iooCkk"
    >
      <p
        class="sc-gtssRu bpiFrI"
      >
        Produtos
        <span>
          R$ 610,00
        </span>
      </p>
      <p
        class="sc-gtssRu bpiFrI"
      >
        Frete
        <span>
          R$ 0,00
        </span>
      </p>
      <p
        class="sc-gtssRu eiTdvO"
      >
        Desconto
        <span>
          R$ 10,00
        </span>
      </p>
      <p
        class="sc-gtssRu hiEsPq"
      >
        Total
        <span>
          R$ 600,00
        </span>
      </p>
    </div>
  </div>
</body>
`;

exports[`<Summary /> should be able to render correctly when not discount 1`] = `
<body>
  <div>
    <div
      class="sc-bdnylx iooCkk"
    >
      <p
        class="sc-gtssRu bpiFrI"
      >
        Produtos
        <span>
          R$ 610,00
        </span>
      </p>
      <p
        class="sc-gtssRu bpiFrI"
      >
        Frete
        <span>
          R$ 0,00
        </span>
      </p>
      <p
        class="sc-gtssRu hiEsPq"
      >
        Total
        <span>
          R$ 600,00
        </span>
      </p>
    </div>
  </div>
</body>
`;

exports[`<Summary /> should be able to render correctly when not shipping 1`] = `
<body>
  <div>
    <div
      class="sc-bdnylx iooCkk"
    >
      <p
        class="sc-gtssRu bpiFrI"
      >
        Produtos
        <span>
          R$ 610,00
        </span>
      </p>
      <p
        class="sc-gtssRu eiTdvO"
      >
        Desconto
        <span>
          R$ 10,00
        </span>
      </p>
      <p
        class="sc-gtssRu hiEsPq"
      >
        Total
        <span>
          R$ 600,00
        </span>
      </p>
    </div>
  </div>
</body>
`;

exports[`<Summary /> should be able to render correctly when not shipping and discount 1`] = `
<body>
  <div>
    <div
      class="sc-bdnylx iooCkk"
    >
      <p
        class="sc-gtssRu bpiFrI"
      >
        Produtos
        <span>
          R$ 610,00
        </span>
      </p>
      <p
        class="sc-gtssRu hiEsPq"
      >
        Total
        <span>
          R$ 600,00
        </span>
      </p>
    </div>
  </div>
</body>
`;
